Shell/SymCounter.cpp

```diff
@@ -196,9 +196,16 @@ void SymCounter::count(Literal* l,int polarity,int add)
   _preds[pred].add(l->isPositive() ? polarity : -polarity,add);
 
   if (!l->shared()) {
-    NonVariableIterator nvi(l);
-    while (nvi.hasNext()) {
-      count(nvi.next().term(), 1, add);
+    for(TermList* arg=l->args(); arg->isNonEmpty(); arg=arg->next()) {
+      if(arg->isTerm()){
+        count(arg->term(), 1, add);
+      }
+    }
+    if(l->isTwoVarEquality()){
```

Comment:
Why is this special-cased exactly? I'm sure it makes sense but I should probably figure it out.

Reply:
In the two variable equality case, we cannot work out the sort of the variables, so have to store it as a separate field in the literal. In the polymorphic case, this field may be a complex term and therefore needs scanning.

Reply:
Aha, OK. I will keep an eye out for other such instances then.